server/users.ts

```diff
@@ -1034,6 +1009,32 @@ export class User extends Chat.MessageContext {
 		}
 		this.updateReady(connection);
 	}
+	getRooms() {
+		const rooms = [];
+		for (const curRoom of Rooms.rooms.values()) {
+			if (this.inRoom(curRoom)) rooms.push(curRoom);
+		}
+		return rooms;
+	}
+	inRoom(roomid: RoomID | Room | BasicRoom) {
+		const room = Rooms.get(roomid);
+		if (!room) return false;
+		return this.id in room.users;
+	}
+	getGames() {
+		const games: RoomGame[] = [];
+		for (const curRoom of Rooms.rooms.values()) {
+			if (this.inGame(curRoom)) {
+				games.push(curRoom.game!);
+			}
+		}
+		return games;
+	}
+	inGame(roomid: RoomID | Room) {
+		const room = Rooms.get(roomid);
+		if (!room) return false;
+		return !!room.game?.playerTable[this.id];
```

Comment:
```suggestion
		if (!room || !room.game) return false;
		return !!room.game.playerTable[this.id];
```

Inner loop, so we might as well avoid the `?.`